### PR TITLE
Support for setting the remote address of a transaction via the public api

### PR DESCRIPTION
**Pull request by tobiasstadler**
## What does this PR do?
Adds support for setting the remote address of a transaction via the public api

## Checklist
- [x] This is an enhancement of existing features, or a new feature in existing plugins
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [x] Added an API method or config option? Document in which version this will be introduced
  - [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: apm-agent-api/src/main/java/co/elastic/apm/api/Transaction.java [hunk 1]
```diff
@@ -229,6 +229,15 @@
      */
     Transaction setUser(String id, String email, String username, String domain);
 
+    /**
+     * Provides a way to manually set the {@code http.request.socket.remote_address} field.
```
Comment (tobiasstadler) [inline apm-agent-api/src/main/java/co/elastic/apm/api/Transaction.java:233]: Maybe `http.request.socket.remote_address` should be `client.source.ip` (see elastic/apm-server#6152)?

**Comment by tobiasstadler** [comment]
ping

**Comment by tobiasstadler** [comment]
ping

**Comment by tobiasstadler** [comment]
superseded by #2513

**Comment by tobiasstadler** [comment]
ping

**Comment by tobiasstadler** [comment]
ping

**Comment by jackshirazi** [comment]
Hi @tobiasstadler, could you give a little detail on why this feature addition is useful? We're more focused on enhancing our support of OpenTelemetry so that eventually that provides the API for the agent, but we can add to the existing API if it's useful and simple

**Comment by tobiasstadler** [comment]
I would like to record the "client ip" in my ejb plugin (https://github.com/tobiasstadler/apm-wildfly-remote-ejb-plugin). From time time I have the need to filter my transactions based on the service (which in our case is mostly identical with the client ip) that is calling my service. right now this is hard to accomplish reliably fro me.

**Comment by jackshirazi** [comment]
Thanks. Couple of things. 

We think that if the transaction api is used to set the remote address, then that should take precedence over any discovered address - even if the transaction sets it to null or empty string. I think atm the final value would just be down to the order of setting

Secondly, could you add a test?

**Pull request closed without merging**